src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -17175,5 +17175,421 @@ static void Main(List<int> list)
+                }
+                """;
+
+            var comp = CreateCompilation(source).VerifyEmitDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "42 43 44 3");
+
+            var program = comp.GetMember<NamedTypeSymbol>("C");
+            var argument = program.GetAttributes().Single().ConstructorArguments.Single();
+            var values = argument.Values;
+            Assert.Equal(3, values.Length);
+            Assert.Equal(42, values[0].Value);
+            Assert.Equal(43, values[1].Value);
+            Assert.Equal(44, values[2].Value);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69133")]
+        public void InAttribute_Params()
+        {
+            string source = """
+                using System;
+
+                var attr = (XAttribute)System.Attribute.GetCustomAttribute(typeof(C), typeof(XAttribute));
+                Console.Write($"{attr._values[0]} {attr._values[1]} {attr._values[2]} {attr._values.Length}");
+
+                [X([42, 43, 44])]
+                class C
+                {
+                }
+
+                public class XAttribute : System.Attribute
+                {
+                    public int[] _values;
+                    public XAttribute(params int[] values) { _values = values; }
+                }
+                """;
+
+            var comp = CreateCompilation(source).VerifyEmitDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "42 43 44 3");
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69133")]
+        public void InAttribute_StringConstants()
+        {
+            string source = """
+                using System;
+
+                var attr = (XAttribute)System.Attribute.GetCustomAttribute(typeof(C), typeof(XAttribute));
+                Console.Write($"{attr._values[0]} {attr._values[1] is null} {attr._values.Length}");
+
+                [X(["hi", null])]
+                class C
+                {
+                }
+
+                public class XAttribute : System.Attribute
+                {
+                    public string[] _values;
+                    public XAttribute(string[] values) { _values = values; }
+                }
+                """;
+
+            var comp = CreateCompilation(source).VerifyEmitDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "hi True 2");
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69133")]
+        public void InAttribute_NestedArray()
+        {
+            string source = """
+                [X([[1], [2]])]
+                class C
+                {
+                }
+
+                public class XAttribute : System.Attribute
+                {
+                    public XAttribute(int[][] values) { }
+                }
+                """;
+
+            CreateCompilation(source, targetFramework: TargetFramework.Net70).VerifyEmitDiagnostics(
```

Comment:
>targetFramework: TargetFramework.Net70 [](http://example.com/codeflow?start=38&length=38)

Is there a reason to use `TargetFramework.Net70` for this test and several below?